src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateDublicate = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    const { extraData, keysToRemove } = action;
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateDublicate, extraData);
+        break;
+
+      case 'removeProperties':
+        keysToRemove.forEach(prop => delete stateDublicate[prop]);
```

Comment:
Don't slice variables name: prop => property.
On my opinion, better to write code inside { } for cases.